internal/pkg/template/artifactpath/artifactpath.go

```diff
@@ -0,0 +1,41 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package artifactpath holds functions to generate the S3 object path for artifacts.
+package artifactpath
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"path"
+)
+
+const (
+	s3ArtifactDirName         = "manual"
+	s3TemplateDirName         = "templates"
+	s3ArtifactAddonsDirName   = "addons"
+	s3ArtifactEnvFilesDirName = "env-files"
+)
+
+// MkdirSHA prefixes the key with the SHA256 hash of the contents of "manual/<hash>/key".
+func MkdirSHA256(key string, content []byte) string {
+	return path.Join(s3ArtifactDirName, fmt.Sprintf("%x", sha256.Sum256(content)), key)
+}
+
+// Addons returns the path to store addon artifact files.
+// Example: manual/addons/key/sha.yml.
+func AddonsWithSHA256(key string, content []byte) string {
```

Comment:
nit: It should be fine to drop the WithSHA256 suffix now that we do it for everything